test/pylib/minio_server.py

```diff
@@ -176,6 +164,33 @@ async def start(self):
 
             await asyncio.sleep(0.1)
 
+        return cmd
+
+    async def start(self):
+        if self.srv_exe is None:
+            self.logger.info("Minio not installed, get it from https://dl.minio.io/server/minio/release/linux-amd64/minio and put into PATH")
+            return
+
+        self.log_file = self.log_filename.open("wb")
+        os.mkdir(self.rootdir)
+
+        while self.cmd is None:
+            self.port = await self._find_available_port(self.port, self.port + 1000)
```

Comment:
ֹUnless I misunderstand something, this can be an infinite loop. If self.port reaches 65536, all the next iterations will do a lot of work but fail with absolute certainty.
I agree with @gusev-p that just trying a random port will be easier. Please make sure to respect /proc/sys/net/ipv4/ip_local_port_range when picking a random port (or just assume some reasonably-high lower range - don't start at 0).